dbt/adapters/redshift/connections.py

```diff
@@ -38,33 +40,176 @@ class RedshiftConnectionMethod(StrEnum):
 
 
 @dataclass
-class RedshiftCredentials(PostgresCredentials):
+class RedshiftCredentials(Credentials):
+    host: str
+    user: str
+    port: Port
     method: str = RedshiftConnectionMethod.DATABASE  # type: ignore
     password: Optional[str] = None  # type: ignore
     cluster_id: Optional[str] = field(
         default=None,
         metadata={"description": "If using IAM auth, the name of the cluster"},
     )
     iam_profile: Optional[str] = None
-    iam_duration_seconds: int = 900
-    search_path: Optional[str] = None
-    keepalives_idle: int = 4
+    iam_duration_seconds: Optional[
+        int
+    ] = 900  # TODO:Not supported in redshift python connector, should we remove this?
```

Comment:
if it doesn't support these options is it because it's no longer needed or because the functionality isn't supported?

Reply:
This is obfuscated from the users of redshift python connector, it's managed by redshift connector whenever we want to use IAM profile as the mode of authn. In essence, no longer needed.